sdk/log/record.go

```diff
@@ -41,6 +41,20 @@ func putIndex(index map[string]int) {
 	indexPool.Put(index)
 }
 
+// seenPool is a pool of boolean maps used for duplicate checking.
+var seenPool = sync.Pool{
+	New: func() any { return make(map[string]bool) },
```

Comment:
```suggestion
	New: func() any { return make(map[string]struct{}) },
```

For such cases, adjusting to a `struct{}` like this might result in lower memory usage.